compiler/rustc_resolve/src/late/diagnostics.rs

```diff
@@ -396,6 +396,18 @@ impl<'a: 'ast, 'ast> LateResolutionVisitor<'a, '_, 'ast> {
             })
             .collect::<Vec<_>>();
         let crate_def_id = CRATE_DEF_ID.to_def_id();
+        // Try to filter out intrinsics candidates, as long as we have
+        // some other candidates to suggest.
+        let intrinsic_candidates: Vec<_> = candidates
+            .drain_filter(|sugg| {
+                let path = path_names_to_string(&sugg.path);
+                path.starts_with("core::intrinsics::") || path.starts_with("std::intrinsics::")
```

Comment:
You could add a `#[diagnostic_item]` attribute to the intrinsics `extern` block and check that the parent of the suggestion is not that specific diagnostic item.

Reply:
do I have access to diagnostic items in rustc_resolve? i don't have a tcx here, i don't think.

Reply:
oh right :D